lib/modules/manager/puppet/puppetfile-parser.ts

```diff
@@ -0,0 +1,79 @@
+import { regEx } from '../../../util/regex';
+import type { PuppetForgeUrl, Puppetfile, PuppetfileModule } from './types';
+
+const lineTerminationRegex = regEx(`\r?\n`);
+const forgeRegex = regEx(/^forge\s+['"]([^'"]+)['"]/);
+
+export function parsePuppetfile(content: string): Puppetfile {
+  const puppetfile: Puppetfile = new Map<PuppetForgeUrl, PuppetfileModule[]>();
+
+  let currentForge = undefined;
+  let currentPuppetfileModule: PuppetfileModule = {};
+
+  for (const rawLine of content.split(lineTerminationRegex)) {
```

Comment:
```suggestion
  for (const rawLine of content.split(newlineRegex)) {
```

import from `util/regex` 😉

Reply:
Thank you!

Actually I copied the regex from the golang datasource. They didn't use the `util/regex.newlineRegex` either:
[.lib/modules/datasource/golang-version/index.ts:8](https://github.com/renovatebot/renovate/blob/main/lib/modules/datasource/golang-version/index.ts#L8)